New post on Ecto.Multi

svileng opened the pull request:
As promised 😄 

I wasn't sure where to put the article, since this is a candidate for the new Ecto section (#1519) but I thought it will be awhile until all lessons are completed, so I added this as a post. Please let me know if that's okay.

Any ideas how to improve the content are also welcome; particularly the `excerpt` is bothering me a bit, it doesn't feel right (I've never been good at writing those), so ideas on this are also appreciated.

doomspork commented:
@svileng thank you so much!  I haven't forgotten your email, I intend to respond I have just been very busy this week.  😁 

I've added some folks so we can get a couple eyes on this.  Very exciting stuff, thank you so much for your involvement and support 😁

gemantzu commented on _posts/2018-10-10-ecto-multi.md:
````diff
@@ -0,0 +1,124 @@
+|> Ecto.Multi.delete(:foo, foo_changeset)
+|> Repo.transaction()
+```
+The atoms used — `:user` ,`:team` and `:foo` — are up to you. You can pass anything (also you can use a string, instead of an atom) as long as it’s unique value for the current Multi.
````
You probably want to add an `a`, making it `as long as it's a unique value for the current Multi.`

svileng commented on _posts/2018-10-10-ecto-multi.md:
```diff
@@ -0,0 +1,124 @@
+If you find yourself running and managing many database queries (and other operations), then keep reading and you may find some useful tools to add your Elixir/Ecto toolbox.
+
+## Creating a Multi
+Everything starts with a `%Multi{}` struct. Regardless of what you’re doing, you always have to provide a new or an existing Multi to most of the functions, which you can easily create by calling `new()`:
```
This sentence is a bit ambiguous:

```
Regardless of what you’re doing, you always have to provide a new or an existing Multi to most of the functions, which you can easily create by calling `new()`
```
Will split to:
```
Regardless of what you’re doing, you always have to provide a new or an existing Multi to most of the helper functions. You can create a new Multi calling the `new()` function:
```

svileng commented on _posts/2018-10-10-ecto-multi.md:
````diff
@@ -0,0 +1,124 @@
+iex> Ecto.Multi.new() |> Repo.transaction()
+{:ok, %{}}
+```
+Clearly we just ran an empty Multi, which was obviously successful since nothing was performed (and nothing returned in the second element of the {:ok, return} tuple. To make Multis useful, you need to add operations to it.
````
Another note to self:
```
you need to add operations to it.
```
should be 
```
you need to add operations to them.

```

svileng commented on _posts/2018-10-10-ecto-multi.md:
````diff
@@ -0,0 +1,125 @@
+```elixir
+Ecto.Multi.new()
+|> Ecto.Multi.insert(:team, team_changeset)
+|> Ecto.Multi.run(:user, fn %{team: team} ->
````
Note: Ecto 3.0 changes. The callback function now 2 params:
```suggestion
|> Ecto.Multi.run(:user, fn _repo, %{team: team} ->
```
This has to be changed in a few places to make this article up to date.

svileng commented:
@doomspork no worries and thank you for the kind words!

Ah yes, will do! Let's hold off from merging until I add that + the Ecto 3.0 changes; I also need to squash all into 1 commit so the git log is nice & clean when we merge.

svileng commented on _data/contributors.yml:
```diff
@@ -348,12 +348,18 @@
   languages:
     - vi
 
-- github: notactuallypagemcconnell
```
Note: I noticed that @notactuallypagemcconnell appears twice in the list, and previous entry is more complete, so I deleted this one (sorry @notactuallypagemcconnell 😆)

svileng commented:
- Update post date;
- Small updates to reflect `Multi.run/3` changes in Ecto 3.0;
- Added myself to contributors.yml
- Squashed all post changes into single commit

doomspork commented:
Thank you, thank you, thank you @svileng!  My apologizes for this taking a bit to get merged.  Things are starting to settle down so I'll be quicker next go round 😁

svileng commented:
@doomspork no worries at all! Let's discuss moving this into the new Ecto section (#1519) once all other Ecto chapters are complete 👍

doomspork commented:
Sounds great @svileng 👍
